src/main/java/Duke.java

```diff
@@ -1,10 +1,271 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+
+
 public class Duke {
+    public static ArrayList<Task> todos = new ArrayList<>();
+    public static File dir = new File("./data/");
+    public static File file = new File("./data/todo_list.txt");
+
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+
+        public void markAsDone() {
+            this.isDone = true;
+        }
+
+        public void markAsNotDone() {
+            this.isDone = false;
+        }
+
+        @Override
+        public String toString() {
+            return "[" + getStatusIcon() + "] " + this.description;
+        }
+    }
+
+    public static class Todo extends Task {
+        public Todo(String description) {
+            super(description);
+        }
+        @Override
+        public String toString() {
+            return "[T]" + super.toString();
+        }
+    }
+
+    public static class Deadline extends Task {
+
+        protected String by;
+
+        public Deadline(String description, String by) {
+            super(description);
+            this.by = by;
+        }
+
+        @Override
+        public String toString() {
+            return "[D]" + super.toString() + " (by: " + by + ")";
```

Comment:
To keep it simple you can use string formatting to prevent accumulation of '+' in string concatenation
```java
int i = 461012;
System.out.format("The value of i is: %d%n", i);
```